torch/csrc/api/include/torch/data/datasets/shared.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <torch/data/datasets/base.h>
+
+#include <memory>
+#include <utility>
+
+namespace torch {
+namespace data {
+namespace datasets {
+
+/// A dataset that wraps another dataset in a shared pointer and implements the
+/// `BatchDataset` API, delegating all calls to the shared instance. This is
```

Comment:
Is the Dataset class example based? If yes, it would be more consistent to call this SharedBatchDataset

Reply:
good point